sw/device/silicon_creator/lib/shutdown.c

```diff
@@ -90,9 +102,10 @@ rom_error_t shutdown_init(lifecycle_state_t lc_state) {
     case kLcStateRma:
       lc_shift = 24;
       break;
-    default:
-      // No alert init for any other lifecycle states.
-      return kErrorOk;
+    default:  // Default to PROD settings.
+      // TODO(lowRISC/opentitan#8288): should an error be reported?
+      lc_shift = 0;
+      break;
```

Comment:
I spoke to tjaychen about the remaining states not covered by the current switch statement.

`kLcStateTestLocked<n>`: we currently aren't expecting code execution in this state.
`kLcStateScrap`: the chip is scrapped, code execution is forbidden.
`kLcStatePostTransition`: the chip has performed a lifecycle transition and needs to be rebooted.
`kLcStateEscalate`: alert handler has signalled to lc_ctrl that something is wrong.
`kLcStateInvalid`: the LC/OTP state is invalid (e.g. has been attacked).

I'm loathe to suggest that we return an error or go to immediate shutdown this early in the boot, but that might be the right thing to do.  For the `TestLocked` states, I'm uncertain.  For `Scrap` - `Invalid`, insta-death seems like the right thing to do.

Reply:
* I agree `kLcStateInvalid` and `kLcStateScrap` should result in shutdown. 
* The other states seem like should be interpreted as errors by the mask ROM - we should confirm with @msfschaffner and @tjaychen. Continuing execution as in `PROD` mode and later checking again the lc_state  and escalating on mismatch seem reasonable as this will add enough redundancy to the boot flow.

The redundancy is probably already covered by `sec_mmio`.

Reply:
One thing to note here is that the life cycle controller disables the CPU in all these states - hence, unless the chip has been tampered with, the CPU will not be able to execute anything / perform a shutdown procedure in these states.

So, with that in mind, any SW handling at this point is a sort of fail safe mechanism.
If it helps to streamline the code, we could probably think about always taking the same action (insta-death sounds like the right thing...).

Reply:
Thanks for the input everyone. I've modified the code so that `shutdown_finalize` is called immediately if the function is provided any of the unhandled lifecycle states Chris listed. I'd also be fine with shutting down after configuring the alert handler if that is preferred.